### PR TITLE
[SPARK-11644][SQL] Remove the option to turn off unsafe and codegen.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

**Comment by rxin** [comment]
cc @yhuai @davies

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 1]
```diff
@@ -56,16 +56,6 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   // sqlContext will be null when we are being deserialized on the slaves.  In this instance
   // the value of codegenEnabled/unsafeEnabled will be set by the desserializer after the
```
Comment (davies) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala]: remove this line

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala [hunk 2]
```diff
@@ -368,8 +367,7 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case logical.Project(projectList, child) =>
         // If unsafe mode is enabled and we support these data types in Unsafe, use the
         // Tungsten project. Otherwise, use the normal project.
-        if (sqlContext.conf.unsafeEnabled &&
-          UnsafeProjection.canSupport(projectList) && UnsafeProjection.canSupport(child.schema)) {
+        if (UnsafeProjection.canSupport(projectList) && UnsafeProjection.canSupport(child.schema)) {
```
Comment (davies) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: UnsafeProjection should support all the dataType now, we could also remove this.

**Comment by davies** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala (hunk 1 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala]: done

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala (hunk 2 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: Yea let's do that in a separate PR. If it does support all types, we should also just remove the normal Project.

**Comment by davies** [comment]
These two tests need fix:
```
 org.apache.spark.sql.execution.metric.SQLMetricsSuite.Project metrics	42 ms	1
 org.apache.spark.sql.hive.execution.HiveExplainSuite.explain extended command
```
